lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -165,6 +169,21 @@ def each_tenant
         each_resource(topology_internal_url_for("tenants")) { |tenant| yield tenant["external_tenant"] }
       end
 
+      # Set of ids for supported applications
+      def supported_application_type_ids
+        ids = []
+        each_resource(sources_api_url_for("application_types")) do |application_type|
```

Comment:
What do you think of letting the server handle the filtering with something like this?
```ruby
supported_applications_filter = SUPPORTED_APPLICATIONS.collect { |sa| "filter["name"][]=#{sa}" }
each_resource("#{sources_api_url_for("application_types")}?#{supported_applications_filter}").collect { |i| i["id"] }
```

Reply:
`supported_applications_filter` could even be a constant, right?

Reply:
Yeah good idea, as for the `.collect` its going to have to wait for the enumerable PR :)

Reply:
Hm this filter doesn't seem to work
`"filter[name][]=/insights/platform/catalog&filter[name][]=/insights/platform/topological-inventory"`

```
url = "http://localhost:3000/api/v1.0/application_types?filter%5Bname%5D%5B%5D%3D%2Finsights%2Fplatform%2Fcatalog%26filter%5Bname%5D%5B%5D%3D%2Finsights%2Fplatform%2Ftopological-inventory"
```

```
[----] I, [2019-08-22T15:03:03.100215 #20613:2b030fb17f00]  INFO -- : Started GET "/api/v1.0/application_types?filter%5Bname%5D%5B%5D%3D%2Finsights%2Fplatform%2Fcatalog%26filter%5Bname%5D%5B%5D%3D%2Finsights%2Fplatform%2Ftopological-inventory" for ::1 at 2019-08-22 15:03:03 -0400
[----] I, [2019-08-22T15:03:03.103222 #20613:2b030fb17f00]  INFO -- : Processing by Api::V1x0::ApplicationTypesController#index as */*
[----] I, [2019-08-22T15:03:03.103333 #20613:2b030fb17f00]  INFO -- :   Parameters: {"filter"=>{"name"=>[{"=/insights/platform/catalog&filter"=>{"name"=>[{"=/insights/platform/topological-inventory"=>nil}]}}]}}
[----] D, [2019-08-22T15:03:03.104479 #20613:2b030fb17f00] DEBUG -- :   Tenant Load (0.3ms)  SELECT  "tenants".* FROM "tenants" WHERE "tenants"."external_tenant" = $1 LIMIT $2  [["external_tenant", "system_orchestrator"], ["LIMIT", 1]]
[----] I, [2019-08-22T15:03:03.110783 #20613:2b030fb17f00]  INFO -- : Completed 500 Internal Server Error in 7ms (ActiveRecord: 1.8ms)


[----] F, [2019-08-22T15:03:03.111537 #20613:2b030fb17f00] FATAL -- :   
[----] F, [2019-08-22T15:03:03.111598 #20613:2b030fb17f00] FATAL -- : TypeError (can't cast ActionController::Parameters):
[----] F, [2019-08-22T15:03:03.111649 #20613:2b030fb17f00] FATAL -- :   
[----] F, [2019-08-22T15:03:03.111716 #20613:2b030fb17f00] FATAL -- : app/controllers/api/v1/mixins/index_mixin.rb:12:in `index'
app/controllers/application_controller.rb:52:in `block (2 levels) in with_current_request'
app/controllers/application_controller.rb:52:in `block in with_current_request'
app/controllers/application_controller.rb:46:in `with_current_request'
```

Reply:
Why `CGI.escape` over `URI.escape`?  The URI escaped string seems to work correctly.

test = "http://localhost:3000/stuff?filter[thing][]=a/b/c"

URI.escape(test)
=> "http://localhost:3000/stuff?filter[thing][]=a/b/c"

CGI.escape(test)
=> "http%3A%2F%2Flocalhost%3A3000%2Fstuff%3Ffilter%5Bthing%5D%5B%5D%3Da%2Fb%2Fc"

Reply:
Rubocop would complain about using `URI.decode` and said to use `CGI.unescape` so I got in the habit of using CGI over URI.  Okay let me try that.

Reply:
Okay that worked, interestingly the application filter worked but not the application_type filter.